src/System.Memory/src/System/Span.cs

```diff
@@ -233,6 +233,210 @@ internal Span(Pinnable<T> pinnable, IntPtr byteOffset, int length)
         }
 
         /// <summary>
+        /// Clears the contents of this span.
+        /// </summary>
+        
+        public void Clear()
+        {
+            if (_length == 0) { return; }
+
+            ref T r = ref DangerousGetPinnableReference();
+            int length = _length;
+
+            // TODO: Remove ugly regions when the right impl has been found
+            #region Naive impl
+            //for (int i = 0; i < length; i++)
+            //{
+            //    Unsafe.Add<T>(ref r, i) = default(T);
+            //}
+            #endregion
+            #region Simple loop unrolling
+            //int i = 0;
+            //for (; i < (length & ~7); i += 8)
+            //{
+            //    Unsafe.Add<T>(ref r, i + 0) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 1) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 2) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 3) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 4) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 5) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 6) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 7) = default(T);
+            //}
+            //for (; i < (length & ~3); i += 4)
+            //{
+            //    Unsafe.Add<T>(ref r, i + 0) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 1) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 2) = default(T);
+            //    Unsafe.Add<T>(ref r, i + 3) = default(T);
+            //}
+            //for (; i < length; i++)
+            //{
+            //    Unsafe.Add<T>(ref r, i) = default(T);
+            //}
+            #endregion
+            #region Unsafe.InitBlockUnaligned fixed impl (until ref versions of InitBlock added)
+            //unsafe
+            //{
+            //    fixed (byte* p = &Unsafe.As<T, byte>(ref r))
+            //    {
+            //        // For `IsReferenceOrContainsReferences`-types we cannot use
+            //        // InitBlock* since this fills byte-wise, which may cause object reference
+            //        // tearing.
+            //        Unsafe.InitBlockUnaligned(p, 0, (UIntPtr)(length * Unsafe.SizeOf<T>()));
+            //    }
+            //}
+            #endregion
+            #region Unsafe.InitBlockUnaligned impl (ref based)
+            // TODO: Is this safe to do for `IsReferenceOrContainsReferences`-types,
+            //       or might it cause reference tearing, partial clear of address?
+            //Unsafe.InitBlockUnaligned(ref r, 0, (UIntPtr)(length * Unsafe.SizeOf<T>()));
+            #endregion
+            #region memset like impl
+            var ptrSize = Unsafe.SizeOf<IntPtr>();
+            int i = 0;
+            ref byte b = ref Unsafe.As<T, byte>(ref r);
+            // TODO: Need to handle pointer sized length...
+            var byteLength = length * Unsafe.SizeOf<T>();
```

Comment:
@jkotas github hid this comment from me (since I pushed a new commit), but yes! absolutely, had the same idea, And since any `IsReferenceOrContainsReference`-type will have memory aligned to pointer size, we don't need to do the weird alignment stuff. Hope to change this tomorrow...